docs/src/cookbooks/boringutils.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: docs
+title:  "Boring Utils Cookbook"
+section: "chisel3"
+---
+
+# Boring Utils Cookbook
+
+```scala mdoc
+import chisel3._
+import chisel3.util.experimental._
+import chisel3.stage.ChiselStage
+```
+
+Chisel has some experimental utilities for generating synthesizable cross module references that "bore" through the
+hierarchy. The underlying cross module connects are handled by FIRRTL's Wiring Transform.
+
+Consider the following example where you want to connect a component in one module to a component in another. Module
+`Constant` has a wire tied to `42` and `Expect` will assert unless connected to `42`:
+```scala mdoc
+import chisel3._
+class Constant extends MultiIOModule {
+  val x = Wire(UInt(6.W))
+  x := 42.U
+}
+class Expect extends MultiIOModule {
+  val y = Wire(UInt(6.W))
+  y := 0.U
+  // This assertion will fail unless we bore!
+  chisel3.assert(y === 42.U, "y should be 42 in module Expect")
+}
+```
+
+We can then connect `x` to `y` using [[BoringUtils]] without modifiying the Chisel IO of `Constant`, `Expect`, or
+modules that may instantiate them. There are two approaches to do this:
+
+1. Hierarchical boring using [[BoringUtils.bore]]
+
+2. Non-hierarchical boring using [[BoringUtils.addSink]]/[[BoringUtils.addSource]]
+
+### Hierarchical Boring
+
+Hierarchical boring involves connecting one sink instance to another source instance in a parent module. Below,
```

Comment:
I'm not quite sure what "instance" means in the above sentence. To me an instance is almost always an instance of a module. Do you mean a wire? Or do you mean a module instance?